### PR TITLE
Remove obsolete tests

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## What is the goal of this PR?

We remove a test that used to test functionality that is no longer supported, namely, `match` queries with no named variables.

## What are the changes implemented in this PR?

`testRoleTypeScopedSpecifically`: The test used to centre around the `match marriage relates spouse;` query at its conception. This is no longer supported, and the updated version is exactly equivalent to the preceding test.
`testQueryParserWithoutGraph`: This test was present in the initial commit, and it looks like it was testing that we can call a static method without referencing its class. Maybe it had a different use at some point prior, but that is lost to history.

### CODE REVIEW

**Comment by grabl** [comment]
### PR Review Checklist

Do not edit the content of this comment. The PR reviewer should simply update this comment by ticking each review item below, as they get completed.

----
**Code**

- [ ] Packages, classes, and methods have a single domain of responsibility.
- [ ] Packages, classes, and methods are grouped into cohesive and consistent domain model.
- [ ] The code is canonical and the minimum required to achieve the goal.
- [ ] Modules, libraries, and APIs are easy to use, robust (foolproof and not errorprone), and tested.
- [ ] Logic and naming has clear narrative that communicates the accurate intent and responsibility of each module (e.g. method, class, etc.).
- [ ] The code is algorithmically efficient and scalable for the whole application.

**Architecture**

- [ ] Any required refactoring is completed, and the architecture does not introduce technical debt incidentally.
- [ ] Any required build and release automations are updated and/or implemented.
- [ ] Any new components follows a consistent style with respect to the pre-existing codebase.
- [ ] The architecture intuitively reflects the application domain, and is easy to understand.
- [ ] The architecture has a well-defined hierarchy of encapsulated components.
- [ ] The architecture is extensible and scalable.

File: java/parser/test/ParserTest.java [hunk 1]
```diff
@@ -282,14 +272,14 @@ public void testConceptVariable() {
...
-        TypeQLMatch exepcted = match(
+        TypeQLMatch expected = match(
                 var("x").sub(var("z")), var("y").sub(var("z")),
                 var("a").isa(var("x")), var("b").isa(var("y")),
                 not(var("x").is("y")),
                 not(var("a").is("b"))
         );
 
-        assertQueryEquals(exepcted, parsed, query);
+        assertQueryEquals(expected, parsed, query);
```
Comment (dmitrii-ubskii) [inline java/parser/test/ParserTest.java:282]: Drive-by typo fix

File: java/parser/test/ParserTest.java [hunk 2]
```diff
@@ -1013,16 +1003,6 @@ public void testDefineRules() {
...
-    @Test
-    public void testQueryParserWithoutGraph() {
-        final String queryString = "match\n" +
-                "$x isa movie;\n" +
-                "get $x;";
-        TypeQLMatch query = parseQuery("match\n" +
-                                               "$x isa movie; get $x;").asMatch();
-        assertEquals(queryString, query.toString());
-    }
-
```
Comment (dmitrii-ubskii) [inline java/parser/test/ParserTest.java:1025]: Almost identical to the very first `testSimpleQuery` test, but doesn't compare with a builder result.

File: java/parser/test/ParserTest.java [hunk 3]
```diff
@@ -151,16 +151,6 @@ public void testRoleTypeScopedGlobally() {
...
-    @Test
-    public void testRoleTypeScopedSpecifically() {
-        String query = "match\n" +
-                "$m relates spouse;";
-        TypeQLMatch parsed = TypeQL.parseQuery(query).asMatch();
-        TypeQLMatch expected = match(var("m").relates("spouse"));
-
-        assertQueryEquals(expected, parsed, query);
-    }
-
```
Comment (dmitrii-ubskii) [inline java/parser/test/ParserTest.java:163]: Identical to the preceding test.

**Review by flyingsilverfin — APPROVED** [review]
